src/box/lua/config/utils/aboard.lua

```diff
@@ -0,0 +1,95 @@
+-- Alert board.
+
+local datetime = require('datetime')
+local log = require('internal.config.utils.log')
+
+local function aboard_set(self, key, alert)
+    assert(alert.type == 'error' or alert.type == 'warn')
+    if alert.type == 'error' then
+        log.error(alert.message)
+    else
+        log.warn(alert.message)
+    end
+    alert.timestamp = datetime.now()
+    self._alerts[key] = alert
+end
+
+local function aboard_get(self, key)
+    return self._alerts[key]
+end
+
+local function aboard_drop(self, key)
+    self._alerts[key] = nil
+    if self._on_drop ~= nil then
+        self._on_drop(self, key)
+    end
+end
+
+local function aboard_clean(self)
+    self._alerts = {}
+end
+
```

Comment:
Interesting point.

There are several thoughts around.

* The `_on_drop` callback receives a key of the alert that is dropped, so it is supposed to be called for each dropped alert. Calling it for all the alerts on `:clean()` seems a bit expensive.
* It adds extra transitions between states. For example, instead of `check_errors` -> `reload_in_progress` we would do `check_errors` -> `ready` -> `reload_in_progress`.
* The alert board is not the single source of truth for the configuration status, it is one of the sources. The warnings/errors presence is used as a supplementary information to decide about the next status on some transitions, but it has no effect on a transition to, say, `reload_in_progress`.

I would add a separate `on_clean` callback if we actually need to react on this action. However, we call `:clean()` right from the code that manages the status, so there is no need now.

Reply:
I am just worried, that there are a lot of status changes happening in different places completely independently:
1) in `config/init.lua`
2) in `_aboard:status()`
3) in `_on_drop()`

Right now, it is quite difficult to understand status state machine from the code as is, and adding another inconstancy (`:clean()` doesn't call the call-back) isn't helping.

Reply:
Actually, is there a different `_on_drop()` function expected? Maybe we can just inline it and simplify everything.

Reply:
> Right now, it is quite difficult to understand status state machine from the code as is

I agree. OTOH, I don't think that changing the status from `aboard:clean()` would change it.

There are two ways how the status is changed:

* Directly from the `config/init.lua` code.
* As a reaction on an external event using the `on_drop()` callback (which is written in `config/init.lua` code too).

At least, all the changes goes over the same code file.

> Actually, is there a different `_on_drop()` function expected? Maybe we can just inline it and simplify everything.

Inline to where?

If we inline it to aboard.lua, it depends on the config -- it is a circular dependency. And it will manage config's status.

If we inline it to `credentials.lua`, it will manage config's status instead of config itself.

I think that the `on_drop` callback variant looks the best.